src/ripple/app/misc/Validations.cpp

```diff
@@ -387,6 +387,35 @@ class ValidationsImp : public Validations
         return ret;
     }
 
+    std::uint32_t getValidationTime (uint256 const& hash, int minValidations)
+    {
+        if (minValidations < 1)
+            minValidations = 1;
+
+        std::vector <std::uint32_t> times;
+
+        {
+            ScopedLockType sl (mLock);
+
+            auto j = findSet (hash);
+            for (auto& it : *j)
+            {
+                if (it.second->isTrusted())
+                    times.push_back (it.second->getSignTime());
+            }
+        }
+
+        if (times.size() < minValidations)
+            return 0;
+
+        if (times.size() == 1)
+            return times[0];
+
+        // Return median signing time
+        std::sort (times.begin (), times.end ());
+        return (times[times.size() / 2] + times[(times.size()-1) / 2]) / 2;
```

Comment:
This is a very nice way to compute the median, I like it. However, as @vinniefalco points out, you could use `std::nth_element` to simplify the code and reduce the computational complexity. We also wouldn't have to worry about overflow. The cost of doing that is to loose the  nice handling of an even number of elements. 

+1 for using `nth_element`.

Using the average would also be simpler than what's here. Do we expect outliers in our data so median makes sense over average?